api/invite.go

```diff
@@ -526,3 +851,357 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
+// @Failure 401 {object} status.Status "Authorization token is missing or does not provide sufficient privileges"
+// @Failure 403 {object} status.Status "Authorization token is invalid"
+// @Failure 409 {object} status.Status "No notification and email sent; User is already an admin"
+// @Failure 422 {object} status.Status "Error when sending the email (probably caused by the mailling service"
+// @Failure 500 {object} status.Status "Internal error while processing the invite, detailled error returned in the body"// @Router /confirm/send/team-invite [put]
+// @security TidepoolAuth
+func (a *Api) UpdateTeamInvite(res http.ResponseWriter, req *http.Request, vars map[string]string) {
+	// By default, the invitee language will be "en" for Englih (as we don't know which language suits him)
+	// In case the invitee is a known user, the language will be overriden in a later step
+	var inviteeLanguage = "en"
+	tokenValue := req.Header.Get(TP_SESSION_TOKEN)
+	token := a.token(res, req)
+	if token == nil {
+		return
+	}
+
+	invitorID := token.UserID
+	if invitorID == "" {
+		res.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	defer req.Body.Close()
+	var ib = &inviteBody{}
+	if err := json.NewDecoder(req.Body).Decode(ib); err != nil {
+		log.Printf("UpdateInvite: error decoding invite to detail %v\n", err)
+		statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_DECODING_INVITE)}
+		a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+		return
+	}
+
+	if ib.Email == "" || ib.TeamID == "" {
+		statusErr := &status.StatusError{Status: status.NewStatus(http.StatusBadRequest, STATUS_ERR_MISSING_DATA_INVITE)}
+		a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+		return
+	}
+	if ib.IsAdmin == "" {
+		ib.IsAdmin = "false"
+	}
+
+	_, team, err := a.userIsTeamAdmin(tokenValue, token.UserID, ib.TeamID, res)
+	if err != nil {
+		return
+	}
+
+	var isAdmin = strings.ToLower(ib.IsAdmin) == "true"
+	if isAdmin {
+		invitedUsr := a.findExistingUser(ib.Email, a.sl.TokenProvide())
+		if invitedUsr != nil && invitedUsr.UserID != "" {
+			if admin, err := a.isTeamAdmin(invitedUsr.UserID, team); admin || err != nil {
+				statusErr := &status.StatusError{Status: status.NewStatus(http.StatusConflict, STATUS_ALREADY_ADMIN)}
+				a.sendModelAsResWithStatus(res, statusErr, statusErr.Code)
+				return
+			}
+		}
+		invite, _ := models.NewConfirmationWithContext(
+			models.TypeMedicalTeamDoAdmin,
+			models.TemplateNameMedicalteamDoAdmin,
+			invitorID,
+			ib.Permissions)
+
+		// if the invitee is already a user, we can use his preferences
+		invite.TeamID = ib.TeamID
+		invite.Email = ib.Email
+		invite.IsAdmin = ib.IsAdmin
+		invite.Status = models.StatusCompleted
+		if invitedUsr != nil {
+			invite.UserId = invitedUsr.UserID
+			// does the invitee have a preferred language?
+			inviteeLanguage = a.getUserLanguage(invite.UserId, res)
+		}
+
+		var member = store.Member{
+			UserID:           invite.UserId,
```

Comment:
what if inviteUsr == nil?

Reply:
this case should not happen, but agree that we have to manage it by returning a 404 error